src/PeriodicTaskRunnerBackgroundService.cs

```diff
@@ -48,22 +48,22 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         {
             // Ensure that we can crate the service. Do this synchronously so that we'll fail-fast no matter the failure mode
             // if the task can't run at all.
-            using (var scope = this.serviceProvider.CreateScope())
+            if(!this.taskFactory.CanResolvePeriodicTask())
             {
-                _ = scope.ServiceProvider.GetRequiredService<TPeriodicTask>();
+                this.logger.LogCritical("{Type} didn't registered in container", typeof(TPeriodicTask));
+                throw new InvalidOperationException($"{typeof(TPeriodicTask)} didn't registered in container");
```

Comment:
I see your argument here that the error message isn't very good - I'm not sure why the better way to resolve this issue isn't just to call `scope.ServiceProvider.GetService<TPeriodicTask> != null` in here and then throw the InvalidOperationException?

Reply:
Sure, it's will also resolve the problem, but I decided to incapsulate requesting services in factory
If I wasn't use factory, I would did as you say